ibm/service/power/data_source_ibm_pi_storage_pools_capacity.go

```diff
@@ -99,22 +99,22 @@ func dataSourceIBMPIStoragePoolsCapacityRead(ctx context.Context, d *schema.Reso
 	if spc.MaximumStorageAllocation != nil {
 		msa := spc.MaximumStorageAllocation
 		data := map[string]interface{}{
-			MaxAllocationSize: msa.MaxAllocationSize,
-			StoragePool:       msa.StoragePool,
-			StorageType:       msa.StorageType,
+			MaxAllocationSize: *msa.MaxAllocationSize,
+			StoragePool:       *msa.StoragePool,
+			StorageType:       *msa.StorageType,
 		}
 		d.Set(MaximumStorageAllocation, flex.Flatten(data))
 	}
 
-	result := make([]map[string]string, 0, len(spc.StoragePoolsCapacity))
+	result := make([]map[string]interface{}, 0, len(spc.StoragePoolsCapacity))
 	for _, sp := range spc.StoragePoolsCapacity {
 		data := map[string]interface{}{
-			MaxAllocationSize: sp.MaxAllocationSize,
+			MaxAllocationSize: *sp.MaxAllocationSize,
 			PoolName:          sp.PoolName,
 			StorageType:       sp.StorageType,
 			TotalCapacity:     sp.TotalCapacity,
 		}
-		result = append(result, flex.Flatten(data))
+		result = append(result, data)
```

Comment:
Just curious any reason for removing flex.Flatten() call?

Reply:
StoragePoolsCapacity does not need to be a map[string]string since it defines all the attributes and their data type. Compare that with MaximumStorageAllocation where those attributes are not defined so they need to be a map[string]string